src/Core/Domain/Customer/ValueObject/CustomerId.php

```diff
@@ -51,18 +53,21 @@ public function __construct($customerId)
     /**
      * @return int
      */
-    public function getValue()
+    public function getValue(): int
     {
         return $this->customerId;
     }
 
     /**
      * @param int $customerId
      */
-    private function assertIntegerIsGreaterThanZero($customerId)
+    private function assertIntegerIsGreaterThanZero(int $customerId): void
     {
-        if (!is_int($customerId) || 0 > $customerId) {
-            throw new CustomerException(sprintf('Customer id %s is invalid. Customer id must be number that is greater than zero.', var_export($customerId, true)));
+        if (0 > $customerId) {
```

Comment:
I don't think `CustomerId` should accept `0` Like the other VOs we should rely on a `NoCustomerId` in this case

Reply:
But if this modification has too many impacts on the scope of this PR I agree we can do it in another one, but right after this one is merged so that we don't forget it 😅

Reply:
https://github.com/PrestaShop/PrestaShop/issues/25946